dynamic_instrumentation/src/instrument.rs

```diff
@@ -123,6 +129,198 @@ fn is_region_or_unsafe_ptr(ty: Ty) -> bool {
     ty.is_unsafe_ptr() || ty.is_region_ptr()
 }
 
+impl<'tcx> Visitor<'tcx> for CheckAddressTakenLocals<'_, 'tcx> {
+    /// Checks the right hand side of each assignment statement for taking the
+    /// address of a local. This may happen explicitly with something like
+    /// std::ptr::addr_of! or implicitly by taking a reference & of a local.
+    /// Instances/locations of address-taking are saved for a later walk over the MIR
+    /// AST.
```

Comment:
It would be good to note here that address-taking occurs only in `Rvalue`s, not in `Operand`s or `Place`s, so that looking at all MIR assignments covers all places where an address might be taken (including the surface-language constructs you call out by name). Maybe change "assignment statement" to "MIR assignment statement" to disambiguate that this isn't referring to surface-language assignment statements.